client/components/SearchInput.js

```diff
@@ -26,10 +26,9 @@ import { fetchWeather, fetchWeatherWeek } from '../store/weatherStore';
 import useCustomStyles from '~/hooks/useCustomStyles';
 
 export const SearchInput = ({ onSelect, placeholder }) => {
-  const [searchString, setSearchString] = useState('');
+  const [searchString, setSearchString] = useState({ name: '' });
```

Comment:
Why are we making it an object instead of string here?

Reply:
Because in there you used the (set timeout) javascript function and it's not correct to do that, it's bad code and I remove that setTimeout and used react hooks(useState) with object for tracking searching name .
Using an object as state provides the flexibility to handle multiple values within a single state variable. This structure can be beneficial when dealing with complex components where different fields or pieces of data need to be managed collectively.